system/Validation/Rules.php

```diff
@@ -436,6 +436,47 @@ public function required_without(
         return true;
     }
 
+    /**
+     * This field is required when any of the other required fields have their expected values present
+     * in the data.
+     *
+     * Example (The special_option field is required when the normal_option,1,2 field has the given value.):
+     *
+     *     required_if[normal_option,1,2]
+     *
+     * @param string|null          $str
+     * @param string|null          $fieldWithValue that we should check if present
+     * @param array<string, mixed> $data           Complete list of field from the form
+     */
+    public function required_if($str = null, ?string $fieldWithValue = null, array $data = []): bool
+    {
+        if ($fieldWithValue === null || $data === []) {
+            throw new InvalidArgumentException('You must supply the parameters: field,expected_values, data.');
+        }
+
+        // Separate fields and expected values
```

Comment:
Please refrain from adding comments that just describes what the code is doing. They're superfluous.

Reply:
OK, noted. i will update later.